packages/core/src/util/assert.ts

```diff
@@ -102,15 +102,27 @@ export function throwError(msg: string, actual?: any, expected?: any, comparison
 
 export function assertDomNode(node: any): asserts node is Node {
   // If we're in a worker, `Node` will not be defined.
-  assertEqual(
-      (typeof Node !== 'undefined' && node instanceof Node) ||
-          (typeof node === 'object' && node != null &&
-           node.constructor.name === 'WebWorkerRenderNode'),
-      true, `The provided value must be an instance of a DOM Node but got ${stringify(node)}`);
+  if (!(typeof Node !== 'undefined' && node instanceof Node) &&
+      !(typeof node === 'object' && node != null &&
+        node.constructor.name === 'WebWorkerRenderNode')) {
+    throwError(`The provided value must be an instance of a DOM Node but got ${stringify(node)}`);
+  }
 }
 
 
 export function assertIndexInRange(arr: any[], index: number) {
-  const maxLen = arr ? arr.length : 0;
-  assertLessThan(index, maxLen, `Index expected to be less than ${maxLen} but got ${index}`);
+  assertDefined(arr, 'Array must be defined.');
+  const maxLen = arr.length;
+  if (index < 0 || index >= maxLen) {
+    throwError(`Index expected to be less than ${maxLen} but got ${index}`);
+  }
 }
+
+
+export function assertOneOf(value: any, ...validValues: any[]) {
+  for (let index = 0; index < validValues.length; index++) {
+    if (validValues.indexOf(value) !== -1) return true;
+  }
```

Comment:
Can it just be this (it looks like we don't need a `for` loop):
```suggestion
  return validValues.indexOf(value) !== -1;
```

Reply:
opps, sorry about that.